x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/textembedding/TextEmbeddingMlNodeServiceTests.java

```diff
@@ -0,0 +1,288 @@
+            ActionListener<Model> modelListener = ActionListener.<Model>wrap(
+                model -> fail("Model parsing should have failed"),
+                e -> assertThat(e, instanceOf(IllegalArgumentException.class))
+            );
+
+            service.parseRequestConfig(randomInferenceEntityId, taskType, settings, Set.of(), modelListener);
+        }
+
+        // Invalid service settings
+        {
+            var service = createService(mock(Client.class));
+            var settings = new HashMap<String, Object>();
+            settings.put(
+                ModelConfigurations.SERVICE_SETTINGS,
+                new HashMap<>(
+                    Map.of(
+                        TextEmbeddingMlNodeServiceSettings.NUM_ALLOCATIONS,
+                        1,
+                        TextEmbeddingMlNodeServiceSettings.NUM_THREADS,
+                        4,
+                        "not_a_valid_service_setting",
+                        randomAlphaOfLength(10)
+                    )
+                )
+            );
+
+            ActionListener<Model> modelListener = ActionListener.<Model>wrap(
+                model -> fail("Model parsing should have failed"),
+                e -> assertThat(e, instanceOf(IllegalArgumentException.class))
+            );
+
+            service.parseRequestConfig(randomInferenceEntityId, taskType, settings, Set.of(), modelListener);
+        }
+    }
+
+    private ActionListener<Model> getModelVerificationActionListener(MultilingualE5SmallMlNodeServiceSettings e5ServiceSettings) {
+        return ActionListener.<Model>wrap(model -> {
+            assertEquals(
+                new MultilingualE5SmallModel(randomInferenceEntityId, taskType, TextEmbeddingMlNodeService.NAME, e5ServiceSettings),
+                model
+            );
+        }, e -> { fail("Model parsing failed " + e.getMessage()); });
+    }
+
+    public void testParsePersistedConfig() {
+
+        // Null model variant
+        {
+            var service = createService(mock(Client.class));
+            var settings = new HashMap<String, Object>();
+            settings.put(
+                ModelConfigurations.SERVICE_SETTINGS,
+                new HashMap<>(
+                    Map.of(TextEmbeddingMlNodeServiceSettings.NUM_ALLOCATIONS, 1, TextEmbeddingMlNodeServiceSettings.NUM_THREADS, 4)
+                )
+            );
+
+            var e5ServiceSettings = new MultilingualE5SmallMlNodeServiceSettings(
+                1,
+                4,
+                TextEmbeddingMlNodeService.MULTILINGUAL_E5_SMALL_MODEL_ID
+            );
+
+            expectThrows(IllegalArgumentException.class, () -> service.parsePersistedConfig(randomInferenceEntityId, taskType, settings));
+
+        }
+
+        // Invalid model variant
+        // because this is a persisted config, we assume that the model does exist, even though it doesn't. In practice, the trained models
+        // API would throw an exception when the model is used
```

Comment:
I want to call out this corner case because the behaviour is a bit weird. I think it is fine to do it this way, but please let me know if you think it should be different.